packages/client-sdk-nodejs/src/batchutils/batch-functions.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  CacheDelete,
+  CacheGet,
+  CacheSet,
+  ICacheClient,
+} from '@gomomento/sdk-core';
+import {
+  BatchDeleteOptions,
+  BatchDeleteResponse,
+  BatchGetOptions,
+  BatchGetResponse,
+  BatchSetOptions,
+  BatchSetResponse,
+  defaultMaxConcurrentRequests,
+  defaultTtlSeconds,
+} from './batch-props';
+import {range} from '@gomomento/sdk-core/dist/src/internal/utils';
+
+export {
+  BatchDeleteOptions,
+  BatchDeleteResponse,
+  BatchGetOptions,
+  BatchGetResponse,
+  BatchSetOptions,
+  BatchSetResponse,
+  defaultMaxConcurrentRequests,
+  defaultTtlSeconds,
+};
+
+export async function batchGet(
+  cacheClient: ICacheClient,
+  cacheName: string,
+  keys: Array<string | Uint8Array>,
+  options?: BatchGetOptions
+): Promise<BatchGetResponse> {
+  const maxConcurrentGets = options?.maxConcurrentGets
+    ? options.maxConcurrentGets
+    : Math.min(defaultMaxConcurrentRequests, keys.length);
+
+  const batchGetResults = range(maxConcurrentGets).map((workerId: number) =>
+    getWorker(workerId, cacheClient, cacheName, keys)
+  );
+  const awaitAll = await Promise.all(batchGetResults);
+
+  const batchGetResponse: BatchGetResponse = {};
+  awaitAll.forEach(responses => {
+    Object.assign(batchGetResponse, responses);
+  });
+  return batchGetResponse;
+}
+
+async function getWorker(
+  workerId: number,
+  cacheClient: ICacheClient,
+  cacheName: string,
+  keys: Array<string | Uint8Array>
+): Promise<Record<string, CacheGet.Response>> {
+  const responses: Record<string, CacheGet.Response> = {};
+  while (keys.length) {
+    const cacheKey = keys.pop();
+    if (cacheKey !== undefined) {
+      responses[String(cacheKey)] = await cacheClient.get(cacheName, cacheKey);
+    }
+  }
+  return Promise.resolve(responses);
+}
+
+export async function batchSet(
+  cacheClient: ICacheClient,
+  cacheName: string,
+  keys: Array<string | Uint8Array>,
+  values: Array<string | Uint8Array>,
+  options?: BatchSetOptions
+): Promise<BatchSetResponse> {
+  const maxConcurrentSets = options?.maxConcurrentSets
+    ? options.maxConcurrentSets
+    : Math.min(defaultMaxConcurrentRequests, keys.length);
+  const ttl = options?.ttl ? options.ttl : defaultTtlSeconds;
```

Comment:
One of the reasons we added `BatchSetItem` as a separate interface in go-lang was because with this signature we are assuming that a user will not mess up providing `keys` and `values` out of order. Also, a user might want to provide different TTLs for the items in the batchRequest. Given these reasons, in go we have this interface:

```
type BatchSetItem struct {
	Key   momento.Key
	Value momento.Value
	Ttl   time.Duration
}
``

which is what the BatchSet accepts as a list

```
type BatchSetRequest struct {
	Client            momento.CacheClient
	CacheName         string
	Items             []BatchSetItem
	MaxConcurrentSets int
}
```

Do you think it's worth doing that for a/ consistency, and b/ giving users more flexibility and typing it to avoid ordering errors? @cprice404 would love your thoughts too.

Reply:
Yeah that's a fair callout. Thanks for the suggestion @pratik151192 .

Sorry for the mis-direction on that part in my earlier feedback, @anitarua , but I do like @pratik151192 's suggestion here for adding a `BatchSetItem` interface. The TTL therein should be optional, and you can fall back to the default TTL for the client for any item that it's not set explicitly on. That will allow you to remove the ttl flag from `BatchSetOptions`, as well.